### PR TITLE
test: Code Coverage: Add Edit Mileage Lint #1

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at 0a869b9</samp>

This pull request adds file attachment thumbnails to the transaction details page, improves type safety for locations and cost centers, and cleans up some code in the mileage rates service.

### <samp>🤖 Generated by Copilot at 0a869b9</samp>

> _`CostCenterOptions`_
> _New interface for labels_
> _Spring cleaning the code_

### <samp>🤖 Generated by Copilot at 0a869b9</samp>

*  Add `CostCenterOptions` interface to represent cost center options in the UI ([link](https://github.com/fylein/fyle-mobile-app/pull/2228/files?diff=unified&w=0#diff-27ef7dd1e2ecbd89a5c72dea4f238188eeea1e67398a3351d73c9f3cb80f4c36R1-R6))
*  Import `FileObject` and `Location` models to use in transaction interfaces ([link](https://github.com/fylein/fyle-mobile-app/pull/2228/files?diff=unified&w=0#diff-173c888e51f0c760dd978d56b4f3848c035ae1ee0fca9eb9b91726349040bf1bR1), [link](https://github.com/fylein/fyle-mobile-app/pull/2228/files?diff=unified&w=0#diff-db96e89a4671a6577872d9c95a3722abc4d79e6b3f8443ec4f74cedb81915dbaR2))
*  Add `dataUrls` property to `UnflattenedTransaction` interface to hold partial file objects with data URLs of attachments ([link](https://github.com/fylein/fyle-mobile-app/pull/2228/files?diff=unified&w=0#diff-173c888e51f0c760dd978d56b4f3848c035ae1ee0fca9eb9b91726349040bf1bR26))
*  Change type of `locations` property in `Transaction` interface from `any` to `Location[]` to match backend response ([link](https://github.com/fylein/fyle-mobile-app/pull/2228/files?diff=unified&w=0#diff-db96e89a4671a6577872d9c95a3722abc4d79e6b3f8443ec4f74cedb81915dbaL36-R37))
*  Remove unused import of `tap` from `rxjs/operators` in `MileageRatesService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2228/files?diff=unified&w=0#diff-3c789edd12fc41da591bd9c248fc12c338269f4526c520850f6146a1f33e8813L8-R8))
*  Add return type of `string` to `getReadableRate` method in `MileageRatesService` class ([link](https://github.com/fylein/fyle-mobile-app/pull/2228/files?diff=unified&w=0#diff-3c789edd12fc41da591bd9c248fc12c338269f4526c520850f6146a1f33e8813L80-R80))

## Clickup
https://app.clickup.com/t/85ztk92ck

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

**Review by jayfyle — COMMENTED** [review]
The remaining lint issues fix in the next PR

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 1]
```diff
@@ -2466,7 +2509,11 @@ export class AddEditMileagePage implements OnInit {
 
     await modal.present();
 
-    const { data } = await modal.onDidDismiss();
+    const { data } = (await modal.onDidDismiss()) as {
+      data: {
+        updated: unknown;
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2514]: why is this set as unknown return type? We shouldn't be setting something to unknown if there is a way for us to derive the data type

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 2]
```diff
@@ -2361,15 +2398,17 @@ export class AddEditMileagePage implements OnInit {
...
-              this.fg.value.report &&
+              reportValue.report &&
               (etxn.tx.policy_amount === null || (etxn.tx.policy_amount && !(etxn.tx.policy_amount < 0.0001)))
             ) {
-              reportId = this.fg.value.report.rp.id;
+              reportId = reportValue.report.rp.id;
             }
             return of(this.transactionsOutboxService.addEntryAndSync(etxn.tx, etxn.dataUrls, comments, reportId)).pipe(
-              switchMap((txnData: Promise<any>) => from(txnData)),
+              switchMap((txnData: Promise<unknown>) => from(txnData)),
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2411]: Dont use unknown intentionally. Set the proper type here

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 3]
```diff
@@ -2331,9 +2368,9 @@ export class AddEditMileagePage implements OnInit {
           return throwError(err);
         }
       }),
-      switchMap(({ etxn, comment }: any) =>
+      switchMap(({ etxn, comment }: { etxn: UnflattenedTransaction; comment: string }) =>
         from(this.authService.getEou()).pipe(
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2372]: if eou is not being used, why don't we remove the call to `this.authService.getEou()`

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 4]
```diff
@@ -2106,12 +2141,12 @@ export class AddEditMileagePage implements OnInit {
           return throwError(err);
         }
       }),
-      switchMap(({ etxn, comment }: any) =>
+      switchMap(({ etxn, comment }: { etxn: UnflattenedTransaction; comment: string }) =>
         forkJoin({
           eou: from(this.authService.getEou()),
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2146]: if eou is not being used, why don't we remove the call to `this.authService.getEou()`

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts [hunk 5]
```diff
@@ -114,35 +140,35 @@ export class AddEditMileagePage implements OnInit {
...
 
-  individualProjectIds$: Observable<string[]>;
+  individualProjectIds$: Observable<number[]>;
 
   isProjectsEnabled$: Observable<boolean>;
 
   isCostCentersEnabled$: Observable<boolean>;
 
-  customInputs$: Observable<any>;
+  customInputs$: Observable<CustomInputsOption[] | ExpenseField[]>;
```
Comment (Julias0) [inline src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:167]: We shouldnt have mixed types in this - can you try to clean this up

**Review by Julias0 — CHANGES_REQUESTED** [review]
Added comments

**Review by Dimple16 — CHANGES_REQUESTED** [review]
I’ll review once the lint, unit tests, and the current comments are resolved, please fix and reassign

**Review by Dimple16 — COMMENTED** [review]
resolve conflicts too

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2514]: Fixed in the 2nd PR.

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts (hunk 3 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2372]: Will remove this in the next PR for refactoring

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts (hunk 5 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:167]: Will try but it is the same in add-edit expense as well.

File: src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts (hunk 4 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-mileage/add-edit-mileage.page.ts:2146]: Same, will remove in the next PR.

**Review by jayfyle — COMMENTED** [review]
This is the 1st part in fixing linting issue having only half the fixes. The rest are in the 2nd PR.
Some places still have `unknown` and other unused expressions. Will fix the types in the next refactoring PR.

**Review by Dimple16 — APPROVED** [review]

**Comment by Dimple16** [comment]
@jayfyle there are a couple of unresolved comments here, I've approved the PR as of now since you've mentioned the fixes are in the next PR

**Review by Julias0 — APPROVED** [review]

**Pull request merged**